aio_kavenegar/client.py

```diff
@@ -0,0 +1,182 @@
+import json
+
+from typing import Literal
+
+import httpx
+
+from aio_kavenegar.exceptions import APIException, HTTPException
+from aio_kavenegar.types import KavenegarResponse
+
+
+# Default requests timeout in seconds.
+DEFAULT_TIMEOUT: int = 10
+DEFAULT_HOST: str = "api.kavenegar.com"
+DEFAULT_HEADERS: dict = {
+    "Accept": "application/json",
+    "Content-Type": "application/x-www-form-urlencoded",
+    "charset": "utf-8",
+}
+
+
+class AIOKavenegarAPI:
+    """
+    https://kavenegar.com/rest.html
+    """
+
+    def __init__(
+        self,
+        apikey: str,
+        timeout: int = DEFAULT_TIMEOUT,
+        host: str = DEFAULT_HOST,
+        headers: dict = DEFAULT_HEADERS,
+        proxies: dict | None = None,
+        secure: bool = True,
+    ) -> None:
+        """
+        :param str apikey: Kavengera API Key
+        :param int timeout: request timeout, default is 10
+        :param str host: Kavenegar API host, default is `api.kavenegar.com`
+        :param dict headers: headers used when requesting Kavenegar resources, default:
+            {
+                "Accept": "application/json",
+                "Content-Type": "application/x-www-form-urlencoded",
+                "charset": "utf-8",
+            }
+        :param dict proxies: Dictionary mapping protocol to the URL of the proxy:
+            {
+                'http': 'http://192.168.1.10:3128',
+                'https': 'http://192.168.1.10:3129',
+            }
+        """
+        self.version: str = "v1"
+        self.host: str = host
+        self.apikey: str = apikey
+        self.apikey_mask: str = f"{apikey[:2]}********{apikey[-2:]}"
+        self.timeout: int = timeout
+        self.headers: dict = headers
+        if not proxies:
+            self.mounts: dict = None
+        else:
+            self.mounts: dict = {}
+            if http := proxies.get("http"):
+                self.mounts.update({"http://": httpx.HTTPTransport(proxy=http)})
+            if https := proxies.get("https"):
+                self.mounts.update({"https://": httpx.HTTPTransport(proxy=https)})
```

Comment:
I don't personally agree with using the `mounts` param, but if so, the type should be `Optional[dict]` I think.

Reply:
`Optional[dict]` will require an additional check cause it can be `None` and broke `self.mounts.update` as default value provided by dict I think it's OK to keep the type dict.